src/Domain/CoordinateMaps/UniformCylindricalFlatEndcap.cpp

```diff
@@ -0,0 +1,692 @@
+  const size_t num_pts = 100;
+  for (size_t i = 0; i < num_pts; ++i) {
+    const double theta_one = theta_max_one * (i / (num_pts - 1.0));
+    if (not is_invertible_for_target_on_sphere_one(
+            center_one, center_two, radius_one, radius_two, theta_max_one,
+            {{center_one[0] + radius_one * sin(theta_one) * cos(phi),
+              center_one[1] + radius_one * sin(theta_one) * sin(phi),
+              center_one[2] + radius_one * cos(theta_one)}})) {
+      return false;
+    }
+  }
+  return true;
+}
+
+UniformCylindricalFlatEndcap::UniformCylindricalFlatEndcap(
+    const std::array<double, 3>& center_one,
+    const std::array<double, 3>& center_two, double radius_one,
+    double radius_two, double z_plane_one)
+    : center_one_(center_one),
+      center_two_(center_two),
+      radius_one_([&radius_one]() {
+        // The equal_within_roundoff below has an implicit scale of 1,
+        // so the ASSERT may trigger in the case where we really
+        // want an entire domain that is very small.
+        ASSERT(not equal_within_roundoff(radius_one, 0.0),
+               "Cannot have zero radius_one");
+        ASSERT(radius_one > 0.0, "Cannot have negative radius_one");
+        return radius_one;
+      }()),
+      radius_two_([&radius_two]() {
+        // The equal_within_roundoff below has an implicit scale of 1,
+        // so the ASSERT may trigger in the case where we really
+        // want an entire domain that is very small.
+        ASSERT(not equal_within_roundoff(radius_two, 0.0),
+               "Cannot have zero radius_two");
+        ASSERT(radius_two > 0.0, "Cannot have negative radius_two");
+        return radius_two;
+      }()),
+      z_plane_one_(z_plane_one),
+      theta_max_one_([&center_one, &radius_one, &z_plane_one]() {
+        const double cos_theta_max = (z_plane_one - center_one[2]) / radius_one;
+        ASSERT(abs(cos_theta_max) < 1.0,
+               "Plane one must intersect sphere_one, and at more than one "
+               "point. You probably specified a bad value of z_plane_one, "
+               "radius_one, or the z component of center_one.");
+        return acos(cos_theta_max);
+      }()) {
+  // Assumptions made in the map.  Some of these can be relaxed,
+  // as long as the unit test is changed to test them.
+  ASSERT(center_two[2] >= center_one[2] + 1.05 * radius_one,
+         "center_two[2] must be >= center_one[2] + 1.02 * radius_one, not "
+             << center_two[2] << " " << center_one[2] + 1.02 * radius_one);
+  ASSERT(center_two[2] <= center_one[2] + 5.0 * radius_one,
+         "center_two[2] must be <= center_one[2] + 5.0 * radius_one, not "
+             << center_two[2] << " " << center_one[2] + 5.0 * radius_one);
+  ASSERT(radius_two >= 0.1 * radius_one * sin(theta_max_one_),
+         "radius_two is too small: " << radius_two << " "
+                                     << 0.1 * radius_one * sin(theta_max_one_));
```

Comment:
I still think this condition isn't covered in the docs.

However, upon closer inspection, I think this condition is similar to the $\alpha > 1.1 \theta_{1 \mathrm{max}}$ condition because it's a restriction on how small `radius_two` can be in relation to the size of the sphere. I would either add a comment here in the cpp file about it so we don't get confused later on why it's here, or just explicitly state the condition in the docs.